xpf.go

```diff
@@ -0,0 +1,81 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+)
+
+// XPF RR. See https://tools.ietf.org/html/draft-bellis-dnsop-xpf-04
+type XPF struct {
+	Hdr  RR_Header
+	Data XPF_Data `dns:"xpf"` // This is seperated out to make this more compatible with the code generation
```

Comment:
I don’t like the use of a seperate structure like this. Aside from the address fields, this seems easy to parse with the existing generated code.

I think a better approach would be something like `dns:"ip-family:IpVersion"` (bikeshed away; "a/aaaa:IpVersion" maybe) and then add support for that tag to the generator.